### PR TITLE
chore(ci): build and test full docker image

### PR DESCRIPTION
**Pull request by viceice**
hopefully it will build in a timely manner

### CODE REVIEW

**Comment by rarkins** [comment]
What’s the build time increase?

**Comment by ViceIce** [comment]
azure needs ~12 min and circle ~8 min

**Comment by ViceIce** [comment]
i think we can make azure faster, if we copy deps from our docker images (where possible)
like
* go
* elexir
* node & yarn (move to base image)
* docker

**Comment by ViceIce** [comment]
we can also add [openjdk-8-jdk-headless](https://packages.ubuntu.com/bionic-updates/openjdk-8-jdk-headless) to be safe and have a full jdk

**Comment by JamieMagee** [comment]
> we can also add [openjdk-8-jdk-headless](https://packages.ubuntu.com/bionic-updates/openjdk-8-jdk-headless) to be safe and have a full jdk

Yeah, I think it would be safer. Gradle [explicitly specifies](https://docs.gradle.org/current/userguide/installation.html#sec:prerequisites) that they require a JDK

> Gradle runs on all major operating systems and requires only a Java Development Kit version 8 or higher to run.

**Review by JamieMagee — COMMENTED** [review]
Have you tried using `--no-install-recommends` when calling `apt-get`?

**Comment by ViceIce** [comment]
> > we can also add [openjdk-8-jdk-headless](https://packages.ubuntu.com/bionic-updates/openjdk-8-jdk-headless) to be safe and have a full jdk
> 
> Yeah, I think it would be safer. Gradle [explicitly specifies](https://docs.gradle.org/current/userguide/installation.html#sec:prerequisites) that they require a JDK
> 
> > Gradle runs on all major operating systems and requires only a Java Development Kit version 8 or higher to run.

Strange, on debian and ubuntu it only depends on the jre. https://packages.ubuntu.com/bionic-updates/gradle

**Comment by ViceIce** [comment]
maybe we should disable recommends and als upgrades in our ubuntu base image
https://askubuntu.com/questions/887701/avoid-suggests-recommends-and-upgrades-during-apt-get-install

This would reduce our images even more.


@rarkins What do you think?

**Comment by rarkins** [comment]
> maybe we should disable recommends and als upgrades in our ubuntu base image
> [askubuntu.com/questions/887701/avoid-suggests-recommends-and-upgrades-during-apt-get-install](https://askubuntu.com/questions/887701/avoid-suggests-recommends-and-upgrades-during-apt-get-install)
> 
> This would reduce our images even more.
> 
> @rarkins What do you think?

Sounds ok with me

**Comment by ViceIce** [comment]
So, I'll open a pr to the base image, is it open source too?

**Comment by rarkins** [comment]
Yes: https://github.com/renovatebot/docker-base

**Comment by JamieMagee** [comment]
@ViceIce Do you have any comparisons for image size?

**Comment by ViceIce** [comment]
@JamieMagee see #5031

File: azure-pipelines.yml [hunk 1]
```diff
@@ -60,16 +60,12 @@ jobs:
...
+      vmImage: ubuntu-18.04
 
     steps:
-      - bash: |
-          docker pull renovate/renovate:slim
-        displayName: Pull old slim renovate image as cache
-
       - bash: |
           export DOCKER_BUILDKIT=1
-          docker build --cache-from renovate/renovate:slim -t renovate-test -f Dockerfile.slim .
```
Comment (rarkins) [inline azure-pipelines.yml]: How come to remove it?

File: .github/workflows/dockerimages.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: crazy-max/ghaction-docker-buildx@v1
+        with:
+          version: latest
```
Comment (rarkins) [inline .github/workflows/dockerimages.yml]: Should we pin it?

File: .github/workflows/dockerimages.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Build the Docker image
+        run: |
+          docker buildx build \
+            --tag renovate \
+            --file ./${{ matrix.file }} .
+
+      - name: Image history
+        run: docker history renovate
```
Comment (rarkins) [inline .github/workflows/dockerimages.yml]: Any command to easily show us the *total* size? e.g. `docker images | grep renovate` ?

File: azure-pipelines.yml (hunk 1 above)
Comment (ViceIce) [reply azure-pipelines.yml]: It slows down, because the cache does not work as expected

**Comment by ViceIce** [comment]
currently azure is faster, maybe we can in future build the images on master and push cache to github packages, then on pr or branch we can use the cache to build images faster.

with github actions we should have write access to github package registry.

**Comment by JamieMagee** [comment]
I can't see such a difference in speed that we need to worry. Though I can see a benefit for having it all in GitHub Actions e.g. We don't need Azure DevOps, Circle CI, _and_ Docker Hub just to build, package, and publish Renovate.

File: .github/workflows/dockerimages.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: crazy-max/ghaction-docker-buildx@v1
+        with:
+          version: v0.3.1
```
Comment (rarkins) [inline .github/workflows/dockerimages.yml]: How did you determine this version? On Marketplace I see this v1.0.4:

image

**Review by rarkins — COMMENTED** [review]
Verify action version

File: .github/workflows/dockerimages.yml (hunk 4 above)
Comment (ViceIce) [reply .github/workflows/dockerimages.yml]: this specifies the docker buildx version, which is currently [v0.3.1](https://github.com/docker/buildx/releases/latest)

Comment (ViceIce) [reply .github/workflows/dockerimages.yml]: https://github.com/crazy-max/ghaction-docker-buildx#inputs

Comment (rarkins) [reply .github/workflows/dockerimages.yml]: My mistake. I see the `v1` action tag now

**Pull request merged**